window_manager/classes/storage.js

```diff
@@ -55,37 +118,19 @@ export class Storage {
         }));
   }
 
-  /**
-   * Returns actions from the storage.
-   * Note: this method is not performing any validation as the data in the storage
-   *       should be valid.
-   *
-   * @return {Promise<Action[]>}
-   */
-  getActions() {
-    return chrome.storage.sync.get({actions: '[]'}).then((item) => item.actions).then(StorageFromJson.actions);
+  /** @return {Promise<Action[]>} */
+  async getActions() {
```

Comment:
can you avoid async and await here?

return this.validatedConfiguration.then((conf) => config.actions);

Reply:
> can you avoid async and await here?

There is no functional difference.
Most of the time this.validatedConfiguration will have been resolved so the await or then is an effective no-op.

Reply:
Fixed